devnet/cmd/scale-down-ecs-service/main.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"regexp"
+	"syscall"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/jessevdk/go-flags"
+)
+
+var (
+	svc *ecs.ECS
+)
+
+func init() {
+	sess := session.Must(session.NewSessionWithOptions(session.Options{
+		SharedConfigState: session.SharedConfigEnable,
+	}))
+	svc = ecs.New(sess)
+}
+
+func findServiceArns(ctx context.Context, cluster, serviceRegex string) (serviceArns []*string, err error) {
+	r, err := regexp.Compile(serviceRegex)
+	if err != nil {
+		return
+	}
+
+	var lsi = &ecs.ListServicesInput{
+		Cluster: &cluster,
+	}
+	for {
+		var lso *ecs.ListServicesOutput
+		lso, err = svc.ListServicesWithContext(ctx, lsi)
+		if err != nil {
+			return
+		}
+		for _, arn := range lso.ServiceArns {
+			if r.MatchString(*arn) {
+				serviceArns = append(serviceArns, arn)
+			}
+		}
+		if lso.NextToken != nil {
+			lsi.NextToken = lso.NextToken
+		} else {
+			break
+		}
+	}
+	return
+}
+
+func updateServices(ctx context.Context, cluster string, serviceArns []*string) (err error) {
+	for _, serviceArn := range serviceArns {
+		_, err = svc.UpdateServiceWithContext(ctx, &ecs.UpdateServiceInput{
+			Cluster:      &cluster,
+			Service:      serviceArn,
+			DesiredCount: aws.Int64(0),
+		})
+		if err != nil {
+			return
+		}
+	}
+	return
+}
+
+func waitForRunningCount(ctx context.Context, cluster string, serviceArns []*string) (err error) {
+	ticker := time.NewTicker(time.Second * 5)
+main:
```

Comment:
Can you refactor the loop to not use labels? Maybe it's just me, but labels are rarely used in Go and it's just more confusing imo than having a condition to the for loops or breaks

Reply:
how else do you break out of an infinite `for { select { ...`?

Reply:
I kinda like the loop labels, it's something I use quite often when using `for { switch/select`

Reply:
See https://github.com/ChainSafe/gossamer/pull/2096/files/369550ab07a4033df3d2c083d8490b3b47f94841#r766240162